src/bindgen/parser.rs

```diff
@@ -563,6 +572,160 @@ impl Parse {
         nested_modules
     }
 
+    fn load_syn_impl(&mut self, item_impl: &syn::ItemImpl) {
+        let trait_ = if let Some((_, path, _)) = &item_impl.trait_ {
+            Path::new(path.segments.last().unwrap().ident.to_string())
+        } else {
+            return;
+        };
+
+        fn try_type_load(ty: &syn::Type) -> Result<Type, ()> {
+            match Type::load(ty) {
+                Ok(opt) => match opt {
+                    Some(output) => return Ok(output),
+                    None => {
+                        warn!("Err parsing assoc type in impl: Valid but empty type");
+                    }
+                },
+                Err(msg) => {
+                    warn!("Err parsing assoc type in impl: {}", msg);
+                }
+            }
+
+            Err(())
+        }
+
+        for impl_item in &item_impl.items {
+            if let syn::ImplItem::Type(impl_item_type) = impl_item {
+                let ty = if let Ok(ty) = try_type_load(&item_impl.self_ty) {
+                    ty
+                } else {
+                    continue;
+                };
+
+                let ident = impl_item_type.ident.to_string();
+                let ident = Path::new(ident);
+
+                let key = AssocTypeId {
+                    ty: Box::new(ty),
+                    trait_: trait_.clone(),
+                    ident,
+                };
+
+                // Parse concrete type
+                let syn_type = &impl_item_type.ty;
+
+                let mut conc_type = if let Ok(ty) = try_type_load(syn_type) {
+                    ty
+                } else {
+                    continue;
+                };
+
+                self.assoc_types.insert(key.clone(), (conc_type.clone(), 0));
```

Comment:
```suggestion
                let conc_type = self.assoc_types.entry(key.clone()).or_insert((conc_type, 0));
```

there is no need to clone here

Reply:
not that it's no need, but it's actually an error because you have to update the reference in the `assoc_types`

Reply:
move this line into `check_nested_assoc_types` and take `conc_type` by value

Reply:
I tried this but I'm getting similar borrow checker issues. The line of self.assoc_types.entry causes a first mutable borrow and I'm unable to use self.check_nested_assoc_types a few lines lines down as I can only have one mutable borrow at a time

Reply:
yes, I know. You have to find a way. Because cloning here is a logical error